controllers/util/provider.go

```diff
@@ -93,16 +92,22 @@ func encodeAndSignOnboardingToken(privateKeyPath string, ticket services.Onboard
 	return fmt.Sprintf("%s.%s", encodedPayload, encodedSignature), nil
 }
 
-func readAndDecodePrivateKey(privateKeyPath string) (*rsa.PrivateKey, error) {
-	pemString, err := os.ReadFile(privateKeyPath)
+func GetParsedPrivateKey(ctx context.Context, cl client.Client, ns string) (*rsa.PrivateKey, error) {
```

Comment:
"Private Key" is a very generic name. When talking about the entire code base, mentioning "private key" does not help identify what key we are talking about. Please rename it to `LoadOnboardingValidationPrivateKey`

Reply:
renamed the function with suggested name